sqlx-cli/src/bin/cargo-sqlx.rs

```diff
@@ -13,9 +13,12 @@ enum Cli {
 
 #[tokio::main]
 async fn main() {
-    dotenvy::dotenv().ok();
     let Cli::Sqlx(opt) = Cli::parse();
 
+    if !opt.no_dotenv {
+        dotenvy::dotenv().ok();
+    }
```

Comment:
Note: this PR actually regressed default handling of `.env` files because it moves the `dotenv()` call _after_ the Clap parsing, which reads `DATABASE_URL` from the environment.

Releasing a hotfix in #3824.